File: go/FruitIntoBaskets/FruitIntoBaskets.go
package main

import "fmt"

// Time -> 0(N^2) Space -> 0(1)
func totalFruit(fruits []int) int {
	sum := 0

	for idx := range fruits {
		val := idx
		basket1 := -1
		basket2 := -1
		currSum := 0
		for val < len(fruits) {
			if basket1 == -1 {
				basket1 = fruits[val]
				currSum++
			} else if basket2 == -1 {
				basket2 = fruits[val]
				currSum++
			} else if fruits[val] == basket1 || fruits[val] == basket2 {
				currSum++
			} else {
				break
			}
			val++
		}

		if currSum > sum {
			sum = currSum
		}
	}
	return sum
}

func main() {
	fmt.Println(totalFruit([]int{1, 2, 3, 2, 2}))
	fmt.Println(totalFruit([]int{0, 1, 2, 2}))
}

File: python/fruitIntoBaskets.py
# Time -> 0(N^2) Space -> 0(1)
def totalFruit(fruits: list[int]) -> int:
    totalSum = 0

    for i in range(len(fruits)):
        basket1 = -1
        basket2 = -1
        currSum = 0
        for j in range(i, len(fruits)):
            if basket1 == -1:
                basket1 = fruits[j]
                currSum += 1
            elif basket2 == -1:
                basket2 = fruits[j]
                currSum += 1
            elif fruits[j] == basket2 or fruits[j] == basket1:
                currSum += 1
            else:
                break

        if currSum > totalSum:
            totalSum = currSum

    return totalSum


print(totalFruit([1, 2, 3, 2, 2]))
print(totalFruit([0, 1, 2, 2]))
